org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -696,7 +696,12 @@ public boolean setAutoBuilding(boolean enable) throws CoreException {
 							IPath projectFolder = ProjectUtils.getProjectRealFolder(project);
 							Set<String> libraries = preferenceManager.getPreferences().getReferencedLibraries().getInclude();
 							for (String pattern: libraries) {
-								patterns.add(ProjectUtils.resolveGlobPath(projectFolder, pattern).toString());
+								IPath resolved = ProjectUtils.resolveGlobPath(projectFolder, pattern);
+								if (resolved.getDevice() != null) {
+									patterns.add(resolved.toPortableString().replace(resolved.getDevice(), "**"));
```

Comment:
It is unreasonable for device breaks matching in Windows. This hack will break the functionality of Vim. I will investigate this problem after this PR.

Reply:
you mean this will break vim on windows?
what happens if you keep the device?

Reply:
See https://github.com/eclipse/eclipse.jdt.ls/pull/1305#discussion_r357682142, this is not specific to Windows. It is the special character ('*', '?') in glob that will make vim client's watching not work.

> No, if the glob patter is something like `**/Cargo.toml`, it won't be supported in this client.

It is easy to be seen in this piece of code:
```rust
 let recursive_mode = if w.glob_pattern.ends_with("**") { 
     notify::RecursiveMode::Recursive 
 } else { 
     notify::RecursiveMode::NonRecursive 
 }; 
 watcher.watch(w.glob_pattern.trim_end_matches("**"), recursive_mode)?; 
```
The `watcher.watch` only supports listening to a real path, not a glob pattern. The Vim client only handles the special case `<real path>/**` and convert it to listening to a directory.

Reply:
If the device is kept, the file event won't be sent, just as the following code says:

https://github.com/eclipse/eclipse.jdt.ls/blob/cf6358ba4c166fa7d4c45aba5b7b22ea08516524/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ResourceUtils.java#L258-L260

But this is not reasonable, because the glob test in vscode do contain an absolute path in Windows. I would like to investigate into this problem, but in this PR I may just keep the current implementation, because the old lib detecion also replace the device with '**'.

Reply:
@fbricon Well, I root-caused the problem....

Well, in vscode, `C:/...` won't work, but `c:/...` will, because the uri of vscode is `c:%3A/...`, and it seems that the matching is case-sensitive.